api/repositories/service_instance_repository_test.go

```diff
@@ -1016,6 +1016,78 @@ var _ = Describe("ServiceInstanceRepository", func() {
 		})
 	})
 
+	Describe("GetServiceInstanceCredentials", func() {
+		var (
+			secretName string
+			secretData map[string][]byte
+			credential map[string]any
+			secret     *corev1.Secret
+			// serviceInstance *korifiv1alpha1.CFServiceInstance
+			getErr    error
+			decodeErr error
+		)
+
+		BeforeEach(func() {
+			secretName = prefixedGUID("secret")
+
+			_ = createServiceInstanceCR(ctx, k8sClient, secretName, space.Name, "the-service-instance", secretName)
```

Comment:
We tend to avoid using those `createXXXCR` uitlities astheir usage is not readable and puts stuff on the objects that is not directly needed by the test. Instead, we recommend to create the service instace via the k8s client directly. Make sure to check that creation succeeds.